src/hooks/use-rainbow-wallet.hook.ts

```diff
@@ -60,9 +62,10 @@ export const useRainbowWallet = (swapRoutes: CalculatedSwapRoute[]) => {
             return () => {
                 toast.dismiss(toastId);
                 showSuccessToast('Success, you can do the swap.');
+                updateBalances(dispatch, walletAddress);
             };
         }
-    }, [pendingActivationTransaction.data]);
+    }, [pendingActivationTransaction.data, dispatch, walletAddress]);
```

Comment:
on `walletAddress` or `dispatch`, a new toast will be shown.  
We don't need this. Rework this  logic.